scripts/partition_manager.py

```diff
@@ -312,6 +300,11 @@ def resolve(reqs, dp):
             raise PartitionError('Unable to solve the following partitions (endless loop):\n'
                                  + pformat(unsolved))
 
+    if solution[0] != "start" or solution[-1] != "end":
```

Comment:
Would it be easy for you to add a test case for this? To ensure that we don't get regression.

Reply:
There are already quite a few tests for `resolve()` that use `start` and `end`. These would fail here if that didn't work. Is there a specific corner case you want to test? We can't really test-trigger this exception, since it implies a bug in the algorithm. Maybe it makes more sense to use an `assert`?

Reply:
I have misunderstood, I thought this was a bugfix (Y)